### PR TITLE
Added casting int to double for operations with field double in space

### PR DESCRIPTION
**Pull request by wey1and**

### CODE REVIEW

File: src/box/field_def.c [hunk 1]
```diff
@@ -61,7 +61,7 @@ const uint32_t field_mp_type[] = {
 	/* [FIELD_TYPE_STRING]   =  */ 1U << MP_STR,
 	/* [FIELD_TYPE_NUMBER]   =  */ (1U << MP_UINT) | (1U << MP_INT) |
 		(1U << MP_FLOAT) | (1U << MP_DOUBLE),
-	/* [FIELD_TYPE_DOUBLE]   =  */ 1U << MP_DOUBLE,
+	/* [FIELD_TYPE_DOUBLE]   =  */ (1U << MP_DOUBLE) | (1U << MP_UINT) | (1U << MP_INT),
```
Comment (tsafin) [inline src/box/field_def.c:64]: I do understand that that was my patch suggested, but, in any case, I've discovered that during later discussions.

I believe, that in this line we should also allow `MP_FLOAT` data types. 

P.S.
I don't know yet how to generate `MP_FLOAT` messagepack data from generic Tarantool tuple.

File: test/engine/double.test.lua [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+engine = test_run:get_cfg('engine')
+
+test_space_with_double_field = box.schema.space.create('space_with_double', {
+    engine = engine,
+    format = { { 'id', 'unsigned' }, { 'double_field', 'double' } }
+})
+
+test_space_with_double_field:create_index('id', { parts = { 'id' } })
+
+box.space.space_with_double:insert { 1, 1 }
```
Comment (tsafin) [inline test/engine/double.test.lua:12]: I'd use `1.0` also. (It's the same `1` though once kept to Lua number, but regardless)

**Review by tsafin — CHANGES_REQUESTED** [review]
Also, it would be a good idea to put non-empty commit message and pull-request description.

**Comment by wey1and** [comment]
Full implementation requires many unacceptable changes.

**Pull request closed without merging**